pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1888,22 +1897,23 @@ func getUlAddrs(ctx *zedrouterContext,
 		appIPAddr = status.AppIPAddr.String()
 	} else if status.Mac != "" {
 		// XXX or change type of VifInfo.Mac to avoid parsing?
-		mac, err := net.ParseMAC(status.Mac)
+		var mac net.HardwareAddr
+		mac, err = net.ParseMAC(status.Mac)
 		if err != nil {
 			log.Fatal("ParseMAC failed: ", status.Mac, err)
 		}
 		log.Infof("getUlAddrs(%d/%d for %s) app Mac %s\n",
 			ifnum, appNum, netInstStatus.UUID.String(), mac.String())
-		addr, err := lookupOrAllocateIPv4(ctx, netInstStatus, mac)
+		addr, err = lookupOrAllocateIPv4(ctx, netInstStatus, mac)
 		if err != nil {
-			log.Errorf("lookupOrAllocateIPv4 failed %s\n", err)
+			log.Errorf("getUlAddrs: App IP address allocation failed: %s\n", err)
```

Comment:
Shouldn't you return this error as well?

Reply:
The error from lookupOrAllocateIPv4 is already getting returned below. Here we just print the error message.

Reply:
Oh but it is still confusing since normally errors are returned where they are detected, and if I were to look at a log with an error followed by the "done" log message it would seem like the error was not returned.

So its more clear on both parts to return it here.

Reply:
ok

Reply:
Done.